> this still works on Linux -- can you give it a try on Windows and see if it resolves your issue?

Thanks @mbolivar for taking a look at this. Unfortunately no it doesn't work. :(

```
C:\Users\nxa17364\zephyr\samples\hello_world\build>C:/Python36/python.exe C:/Users/nxa17364/zephyr//scripts/support/zephyr_flash_debug.py --verbose jlink flash --board-dir=C:/Users/nxa17364/zephyr/boards/arm/lpcxpresso54114 --kernel-elf=C:/Users/nxa17364/zephyr/samples/hello_world/build/zephyr/zephyr.elf --kernel-hex=C:/Users/nxa17364/zephyr/samples/hello_world/build/zephyr/zephyr.hex --kernel-bin=C:/Users/nxa17364/zephyr/samples/hello_world/build/zephyr/zephyr.bin --gdb=C:/gcc-arm-none-eabi-6_2017-q2-update/bin/arm-none-eabi-gdb.exe --openocd=OPENOCD-NOTFOUND --openocd-search= --dt-flash=y --device=LPC54114J256_M4 --commander=JLink"
Flashing Target Device
JLink -if swd -speed auto -device LPC54114J256_M4 -CommanderScript ''"'"'C:\Users\nxa17364\AppData\Local\Temp\1\tmplq999idcjlink\runner.jlink'"'"''
Traceback (most recent call last):
  File "C:/Users/nxa17364/zephyr//scripts/support/zephyr_flash_debug.py", line 110, in <module>
    main()
  File "C:/Users/nxa17364/zephyr//scripts/support/zephyr_flash_debug.py", line 96, in main
    handlers[args.top_command](args)
  File "C:/Users/nxa17364/zephyr//scripts/support/zephyr_flash_debug.py", line 34, in runner_handler
    runner.run(args.command)
  File "C:\Users\nxa17364\zephyr\scripts\support\runner\core.py", line 394, in run
    self.do_run(command, **kwargs)
  File "C:\Users\nxa17364\zephyr\scripts\support\runner\jlink.py", line 95, in do_run
    self.flash(**kwargs)
  File "C:\Users\nxa17364\zephyr\scripts\support\runner\jlink.py", line 142, in flash
    self.check_call(cmd)
  File "C:\Users\nxa17364\zephyr\scripts\support\runner\core.py", line 435, in check_call
    subprocess.check_call(cmd)
  File "C:\Python36\lib\subprocess.py", line 286, in check_call
    retcode = call(*popenargs, **kwargs)
  File "C:\Python36\lib\subprocess.py", line 267, in call
    with Popen(*popenargs, **kwargs) as p:
  File "C:\Python36\lib\subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "C:\Python36\lib\subprocess.py", line 997, in _execute_child
    startupinfo)
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

--- a/scripts/support/runner/jlink.py
+++ b/scripts/support/runner/jlink.py
@@ -124,15 +124,18 @@ class JLinkBinaryRunner(ZephyrBinaryRunner):
         lines.append('g') # Start the CPU
         lines.append('q') # Close the connection and quit
 
-        with tempfile.NamedTemporaryFile(suffix='.jlink') as f:
-            f.writelines(bytes(line + '\n', 'utf-8') for line in lines)
-            f.flush()
+        # Don't use NamedTemporaryFile: the resulting file can't be
+        # opened again on Windows.
+        with tempfile.TemporaryDirectory(suffix='jlink') as d:
+            fname = os.path.join(d, 'runner.jlink')
+            with open(fname, 'wb') as f:
+                f.writelines(bytes(line + '\n', 'utf-8') for line in lines)
 
             cmd = ([self.commander] +
                    ['-if', self.iface,
                     '-speed', self.speed,
                     '-device', self.device,
-                    '-CommanderScript', f.name])
+                    '-CommanderScript', fname])
 
             print('Flashing Target Device')
             self.check_call(cmd)

